lib/ansible/plugins/lookup/aws_ssm.py

```diff
@@ -96,9 +96,13 @@
 - name: return a dictionary of ssm parameters from a hierarchy path with shortened names (param instead of /PATH/to/param)
   debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', region='ap-southeast-2', shortnames=true, bypath=true, recursive=true ) }}"
 
-- name: Iterate over a parameter hierarchy
-  debug: msg='key contains {{item.Name}} with value {{item.Value}} '
-  loop: '{{ query("aws_ssm", "/TEST/test-list", region="ap-southeast-2", bypath=true) }}'
+- name: Iterate over a parameter hierarchy (one iteration per parameter)
+  debug: msg='Key contains {{ item.key }} , with value {{ item.value }}'
+  loop: '{{ lookup("aws_ssm", "/demo/", bypath=True) | dict2items }}'
```

Comment:
Also no need to remove region from the example.
```suggestion
  loop: '{{ lookup("aws_ssm", "/demo/", region="ap-southeast-2", bypath=True) | dict2items }}'
```